### PR TITLE
build-support: Add fetchfossil function.

### PR DESCRIPTION
**Pull request by MostAwesomeDude**
###### Motivation for this change

I have a Fossil repository that I want to reference from Nix.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is a dead-simple fetcher which clones a Fossil repository, opens it directly into $out, and then nicks out the single Fossil checkout marker.

For some reason, my Let's Encrypt certificates don't work when fetching over HTTPS. I don't know why. It's not a dealbreaker for me, though.

### CODE REVIEW

File: pkgs/build-support/fetchfossil/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{stdenv, fossil}:
+
+{name ? null, url, rev ? null, md5 ? null, sha256 ? null}:
+
+# TODO: statically check if mercurial as the https support if the url starts woth https.
```
Comment (garbas) [inline pkgs/build-support/fetchfossil/default.nix]: i guess this is a leftover from fetchhg right?

Comment (MostAwesomeDude) [inline pkgs/build-support/fetchfossil/default.nix]: Ha, yeah. Lemme fix that up.

**Comment by MostAwesomeDude** [comment]
This is ready for review; I've been using this for a couple days without incident.

File: pkgs/build-support/fetchfossil/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{stdenv, fossil}:
+
+{name ? null, url, rev, md5 ? null, sha256 ? null}:
```
Comment (domenkozar) [inline pkgs/build-support/fetchfossil/default.nix]: We shouldn't support `md5` in newer functions. We're trying to get rid of it. See https://github.com/NixOS/nixpkgs/issues/4491

Comment (MostAwesomeDude) [inline pkgs/build-support/fetchfossil/default.nix]: And good riddance! Changing it to mandate SHA256.

**Comment by domenkozar** [comment]
Looks good. Is there a `.git` equivalent that would change through time (non-deterministic)?

**Comment by MostAwesomeDude** [comment]
No, Fossil repositories are a single SQLite database. The `fossil open` command generates a working tree from the database, and there's just the single `.fslckout` file left in there.

**Pull request merged**